cmd/copy.go

```diff
@@ -1115,7 +1162,7 @@ func (cca *cookedCopyCmdArgs) processRedirectionUpload(blobResource common.Resou
 	}
 
 	// getCredentialInfoForLocation populates oauth token fields... so, it's very easy.
-	credInfo, _, err := getCredentialInfoForLocation(ctx, common.ELocation.Blob(), blobResource.Value, blobResource.SAS, false)
+	credInfo, _, err := getCredentialInfoForLocation(ctx, common.ELocation.Blob(), blobResource.Value, blobResource.SAS, false, cca.cpkOptions)
```

Comment:
> cca.cpkOptions [](start = 124, length = 15)

we didn't call `common.GetClientProvidedKey` yet, any reason for that?

Reply:
Yup. We call `common.GetClientProvidedKey(cpkOptions)` when we need the credentials just before using them. 
If you go inside the definition of this function, we're calling it inside `getBlobCredentialType()`

```
if !isContainer {
	clientProvidedKey := azblob.ClientProvidedKeyOptions{}
	if cpkOptions.IsSourceEncrypted {
		clientProvidedKey = common.GetClientProvidedKey(cpkOptions)
	}
	// Scenario 3: When resourceURL points to a blob
	blobURL := azblob.NewBlobURL(*resourceURL, p)
	if _, err := blobURL.GetProperties(ctx, azblob.BlobAccessConditions{}, clientProvidedKey); err == nil {
		return true
	}
}

```